subprojects/model-core/src/main/java/org/gradle/api/internal/tasks/CachingTaskDependencyResolveContext.java

```diff
@@ -137,5 +141,15 @@ public void getNodeValues(Object node, final Collection<? super T> values, Colle
                 String.format("Cannot resolve object of unknown type %s to a Task.",
                     node.getClass().getSimpleName()));
         }
+
+        private void evaluateCredentialsProvider(Object node) {
+            if (node instanceof ProviderInternal) {
+                ProviderInternal<?> provider = (ProviderInternal<?>) node;
+                Class<?> providedType = provider.getType();
+                if (providedType != null && Credentials.class.isAssignableFrom(providedType)) {
+                    provider.isPresent();
+                }
+            }
```

Comment:
This probably needs better expression/encapsulation, but first I would like to confirm if this is the right place and the right way to intercept credentials provider task properties.